executor/pipeline/stage.rs

```diff
@@ -72,6 +108,10 @@ impl<Snapshot: ReadableSnapshot + 'static> StageIterator for ReadStageIterator<S
         match self {
             ReadStageIterator::Initial(iterator) => iterator.collect_owned(),
             ReadStageIterator::Match(iterator) => iterator.collect_owned(),
+            ReadStageIterator::Sort(iterator) => iterator.collect_owned(),
+            ReadStageIterator::Offset(iterator) => iterator.collect_owned(),
+            ReadStageIterator::Limit(iterator) => iterator.collect_owned(),
+            ReadStageIterator::Filter(iterator) => iterator.collect_owned(),
         }
     }
 }
```

Comment:
Still need to put them in the write stages.

Reply:
Please do it in this PR too!